Mathlib/Topology/Homeomorph.lean

```diff
@@ -766,3 +766,96 @@ def homeoOfEquivCompactToT2 [CompactSpace X] [T2Space Y] {f : X ≃ Y} (hf : Con
     continuous_invFun := hf.continuous_symm_of_equiv_compact_to_t2 }
 
 end Continuous
+
+variable [TopologicalSpace X] [TopologicalSpace Y] [TopologicalSpace Z]
+  {W : Type*} [TopologicalSpace W] {f : X → Y}
+
+/-- Predicate saying that `f` is a homeomorphism. -/
```

Comment:
This docstring should be greatly expanded with information about why we have this, and when to use it, especially since we frequently should prefer `Homeomorph` directly.

For example, this should mention at least:

1. That `Homeomorph` exists and that it should get preferred whenever the inverse is easy to write down.
2. That we have this in part because we have much of the topological end of the library unbundled.
3. Unlike many other `EquivLike` morphisms, `IsHomeomorph` is not just a `FunLike` version of the morphism + `Function.Bijective`, so something is actually gained by having this as a separate predicate.
4. This predicate is useful when you *don't* have an easily written inverse, but you still want to access properties of homeomorphosms or potentially build one.

Points 2 and 3 also explain why we don't have this pattern for various equivs across the library.

Reply:
What about now? I'm not super happy with it, so feel free to request more changes

Reply:
I think it's okay. It may not be perfect, but I don't see obvious bits to change, and it's good enough.